Sources/Sentry/SentryProfiler.mm

```diff
@@ -232,6 +241,26 @@ - (BOOL)isRunning
     return _samplingProfiler->isSampling();
 }
 
+- (void)transmitChunkEnvelope
+{
+    const auto stateCopy = [self.state copyProfilingData];
+    const auto startSystemTime = self.continuousChunkStartSystemTime;
+    const auto endSystemTime = SentryDependencyContainer.sharedInstance.dateProvider.systemTime;
+    self.continuousChunkStartSystemTime = endSystemTime + 1;
+    [self.state clear]; // !!!: profile this to see if it takes longer than one sample duration
+                        // length: ~9ms
+
+    const auto envelope = sentry_continuousProfileChunkEnvelope(
+        startSystemTime, endSystemTime, stateCopy, self.profilerId,
+        [self.metricProfiler serializeBetween:startSystemTime and:endSystemTime]
+#    if SENTRY_HAS_UIKIT
+        ,
+        self.screenFrameData
+#    endif // SENTRY_HAS_UIKIT
+    );
+    [SentrySDK captureEnvelope:envelope];
```

Comment:
`m`: Does it make sense to move this to a background thread to potentially unblock the calling thread, which could be the main thread to tackle https://github.com/getsentry/sentry-cocoa/issues/3893? Also, okay to do this in another PR.

Reply:
Yes, I'm definitely on board to do that, but agree it is appropriate to do so in another PR. ([link](https://www.notion.so/sentry/Continuous-Profiling-Cocoa-SDK-9d64a809ed2d487b89e4aa0358bfac18?pvs=4#5454422467754884b59f496ddaa77b0b))